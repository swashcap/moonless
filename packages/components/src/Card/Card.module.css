/** Card */

.Card {
  background: var(--component-card-background-color);
  border-color: var(--component-card-border-color);
  border-radius: var(--component-card-border-radius);
  border-style: solid;
  border-width: var(--component-card-border-width);
  padding: var(--component-card-padding);
}
