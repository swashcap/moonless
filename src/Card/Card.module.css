.Card {
  border: 1px solid gray;
  padding: 1em;
}
