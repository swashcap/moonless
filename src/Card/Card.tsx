import React from 'react';

import { clsx } from '../utils/clsx';
import styles from './Card.module.css';

export type CardProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

export const Card: React.FC<CardProps> = ({ className, ...rest }) => (
  <div className={clsx(styles.Card, className)} {...rest} />
);
