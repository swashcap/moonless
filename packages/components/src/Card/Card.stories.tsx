import { Card } from './Card';

export default {
  component: Card,
  title: 'Component/Card',
};

export const Default = () => <Card>Default</Card>;
