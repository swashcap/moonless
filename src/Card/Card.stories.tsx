import React from 'react';

import { Card } from './Card';

export default {
  component: Card,
  title: 'Card',
};

export const Default = () => <Card>Default</Card>;
