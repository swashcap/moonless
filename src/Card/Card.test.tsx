import React from 'react';
import { render } from '@testing-library/react';

import { Card } from './Card';

describe('Card', () => {
  test('It should render.', () => {
    const { container } = render(<Card>Card</Card>);

    expect(container).toMatchSnapshot();
  });
});
