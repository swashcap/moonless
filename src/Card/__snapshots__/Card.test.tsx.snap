// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card It should render. 1`] = `
<div>
  <div
    class="Card"
  >
    Card
  </div>
</div>
`;
